core/include/gnuradio-4.0/Graph.hpp

```diff
@@ -806,26 +864,30 @@ struct Graph {
     template<std::size_t sourcePortIndex, std::size_t sourcePortSubIndex, typename Source>
     [[nodiscard, deprecated("For internal use only, the connect with the port name should be used")]] auto
     connect(Source &source) {
-        auto &port_or_collection = outputPort<sourcePortIndex>(&source);
+        auto &port_or_collection = outputPort<sourcePortIndex, traits::port::port_kind::Any>(&source);
         return SourceConnector<Source, std::remove_cvref_t<decltype(port_or_collection)>, sourcePortIndex, sourcePortSubIndex>(*this, source, port_or_collection);
     }
 
     template<std::size_t sourcePortIndex, typename Source>
     [[nodiscard, deprecated("For internal use only, the connect with the port name should be used")]] auto
     connect(Source &source) {
-        return connect<sourcePortIndex, meta::invalid_index, Source>(source);
+        if constexpr (sourcePortIndex == meta::default_message_port_index) {
+            return SourceConnector<Source, decltype(source.builtinMessagePorts.output), meta::invalid_index, meta::invalid_index>(*this, source, source.builtinMessagePorts.output);
+        } else {
+            return connect<sourcePortIndex, meta::invalid_index, Source>(source);
+        }
     }
 
     // connect using the port name
 
     template<fixed_string sourcePortName, std::size_t sourcePortSubIndex, typename Source>
     [[nodiscard]] auto
     connect(Source &source) {
-        using source_output_ports             = typename traits::block::output_ports<Source>;
+        using source_output_ports             = typename traits::block::all_output_ports<Source>;
```

Comment:
what about redining this from `traits::block::all_output_ports` to `(gr::)port::output::all<...>`?

Reply:
This would keep it a bit more brief and similar also with e.g.
 * `port::output::message::all`
 * `port::output::stream::all`
  * ...
  
  :thinking: